feat(github): use repo url to get token (#17179)
renovatebot/renovate: docs/usage/getting-started/private-packages.md

Renovate applies theses `hostRules` to every HTTP(s) request which is sent, so they are largely independent of any platform or datasource logic.
With `hostRules` in place, private package lookups should all work.

### GitHub (and Enterprise) repo scoped credentials

If you need to use seperate credentials for a single GitHub repo which is different from the default, then you can configure `hostRules` like this:

```json
{
  "hostRules": [
    {
      "matchHost": "https://api.github.com/repos/org/repo",
      "token": "abc123"
    },
    {
      "matchHost": "https://gihub.domain.com/api/v3/repos/org/repo",
      "token": "abc123"
    }
  ]
}
```

> Then renovate will use that credentials for all requests which relates to `org/repo`.

Reviewer: ```suggestion
Renovate will use those credentials for all requests to `org/repo`.
```

> Here's a sample for `gomod` with private github.com repos (assume this config is used on `github.com/some-other-org` repo):

Reviewer: ```suggestion
Here's an example for `gomod` with private github.com repos.
Assume this config is used on the `github.com/some-other-org` repo:
```

Renovate applies theses `hostRules` to every HTTP(s) request which is sent, so they are largely independent of any platform or datasource logic.
With `hostRules` in place, private package lookups should all work.

### GitHub (and Enterprise) repo scoped credentials

> If you need to use seperate credentials for a single GitHub repo which is different from the default, then you can configure `hostRules` like this:

Reviewer 2: ```suggestion
If you need to use different credentials for a specific GitHub repo, then you can configure `hostRules` like one of the following:
```

```json
{
  "hostRules": [
    {
      "matchHost": "https://api.github.com/repos/org/repo",
      "token": "abc123"
    },
    {

>       "matchHost": "https://gihub.domain.com/api/v3/repos/org/repo",

Reviewer 2: ```suggestion
      "matchHost": "https://github.domain.com/api/v3/repos/org/repo",
```

      "token": "abc123"
    }
  ]
}
```

Renovate will use those credentials for all requests to `org/repo`.

> 
> Here's an example for `gomod` with private github.com repos.

Reviewer: ```suggestion
#### Example for gomod

Here's an example for `gomod` with private github.com repos.
```

How about putting a new heading here? It helps separate the general explanation and the specific example.

      "token": "abc123"
    },
    {
      "matchHost": "https://github.domain.com/api/v3/repos/org/repo",
      "token": "abc123"
    }
  ]
}
```

Renovate will use those credentials for all requests to `org/repo`.

> #### Example for gomod

Reviewer: ```suggestion

#### Example for gomod
```

Add a newline to fix the lint.
